src/charm.py

```diff
@@ -60,6 +68,32 @@ def __init__(self, *args):
             self.unit.get_container(CONTAINER), User(name=PEBBLE_USER[0], group=PEBBLE_USER[1])
         )
 
+        self.auth_proxy = AuthProxyRequirer(self, self.auth_proxy_config, OATHKEEPER_REL)
+        self.framework.observe(
+            self.auth_proxy.on.auth_proxy_relation_removed, self._on_auth_proxy_removed
+        )
+
+    @property
+    def auth_proxy_config(self) -> Optional[AuthProxyConfig]:
+        """Configure the auth proxy relation."""
+        if self.ingress.url is not None:
```

Comment:
this will cover the case where ingress url is empty
```suggestion
        if self.ingress.url:
```